### DISCUSSION TITLE
How to actually use the open api gradle plugin?

### DISCUSSION THREAD

**Discussion by raistma** [smallrye/smallrye-open-api#1729] [Q&A]
Hi guys, I'm not a gradle expert but I need to use the open api gradle plugin in quarkus app.
The problem is that the moment I add the gradle plugin (io.smallrye.openapi) the build immediately fails with this error:

> Cannot convert provided value (task ':generateOpenApiSpec' property 'outputDirectory') to a file.

I tried overwriting the value in build file but that didn't help.
Do you know what could be the issue?
Could you please provide more detailed readme for the gradle plugin? It seems a bit lacking compared to the maven plugin.

Thanks a lot for your work!

**Comment by MikeEdgar** [comment]
@snazy any chance you have an example handy that could be added to the README? I am usually a bit slow to ramp up working with Gradle :rofl:

**Comment by snazy** [comment]
Sure - we use it daily. Best example is [probably here](https://github.com/projectnessie/nessie/blob/6a97e9e2510b64771950d3ec66ee3ae78b1de416/api/model/build.gradle.kts#L66-L101). We use the generated openapi files also later in a Quarkus server [here](https://github.com/projectnessie/nessie/blob/6a97e9e2510b64771950d3ec66ee3ae78b1de416/servers/quarkus-server/build.gradle.kts#L77) and [here](https://github.com/projectnessie/nessie/blob/6a97e9e2510b64771950d3ec66ee3ae78b1de416/servers/quarkus-server/build.gradle.kts#L117-L124)

**Reply by codespearhead** [reply]
@raistma Does this answer your question? If so, can you [mark this discussion as answered](https://docs.github.com/en/discussions/managing-discussions-for-your-community/moderating-discussions#marking-a-comment-as-an-answer)?
